xbmc/addons/Addon.cpp

```diff
@@ -350,16 +400,42 @@ bool CAddon::SettingsToXML(CXBMCTinyXML &doc) const
   return true;
 }
 
-std::shared_ptr<CAddonSettings> CAddon::GetSettings()
+bool CAddon::InitSettings(uint32_t instance)
 {
   // initialize addon settings if necessary
-  if (m_settings == nullptr)
+  if (m_settings.find(instance) == m_settings.end() || !m_settings[instance].m_control)
   {
-    m_settings = std::make_shared<CAddonSettings>(enable_shared_from_this::shared_from_this());
-    LoadSettings(false);
+    CSettingsData data;
+
+    data.m_control = std::make_shared<CAddonSettings>(enable_shared_from_this::shared_from_this());
+    if (instance == ADDON_SETTINGS_ID)
+    {
+      data.m_baseSettingsPath =
+          URIUtils::AddFileToFolder(m_addonInfo->Path(), "resources", "settings.xml");
+      data.m_userSettingsPath = URIUtils::AddFileToFolder(m_profilePath, "settings.xml");
+    }
+    else
+    {
+      data.m_baseSettingsPath =
```

Comment:
maybe you should rename this member to ``m_addonSettingsPath`? I do remember you renamed other parts from 'base' to 'addon' as well.